code/modules/projectiles/projectile/change.dm

```diff
@@ -54,3 +54,45 @@
 			var/mob/living/carbon/human/H = new_mob
 			to_chat(new_mob, "<B>Your form morphs into that of a [(istype(H) && H.species && H.species.name) ? H.species.name : randomize].</B>")
 			return new_mob
+
+/obj/item/projectile/polymorph
+	name = "bolt of polymorph"
+	icon_state = "ice_1"
+	damage = 0
+	nodamage = TRUE
+	flag = "energy"
+	fire_sound = 'sound/weapons/radgun.ogg'
+	var/status = MAJOR
+
+/obj/item/projectile/polymorph/on_hit(var/atom/hit)
+	if(hit == firer)
+		firer.transmogrify(/mob/living/simple_animal/pollywog, allow_revert = FALSE)
+		return
+	var/stat = status
+	spawn(1)//fixes bugs caused by the target ceasing to exist before the projectile has died.
```

Comment:
Could this be fixed by manually calling whatever proc kills the projectile right here, before running `polymorph()`?